plugins/outputs/influxdb_v2/influxdb.go

```diff
@@ -164,37 +176,53 @@ func (i *InfluxDB) Write(metrics []telegraf.Metric) error {
 			return nil
 		}
 
+		if apiError, ok := err.(*BucketNotFoundError); ok {
+			if i.CreateBuckets {
+				if err := client.CreateBucket(ctx, apiError.Bucket); err != nil {
+					i.Log.Errorf("When writing to [%s]: bucket %q not found and failed to recreate", client.URL(), apiError.Bucket)
+				}
```

Comment:
Do we need to retry the write of the current batch of metrics once we create the bucket? If we don't, the batch is lost for the client.

Reply:
I'm a bit confused by this. It automatically retries the write since an error is returned when the bucket does not exist. Are you suggesting we call `client.Write` again once the bucket is created successfully instead of waiting for the next flush interval?

Reply:
Yes, I think we should call client.Write again right after the bucket is created.

The current code tries to write to each node in the cluster in random sequence. If it fails to write to one node because the bucket doesn't exist, it creates the bucket on that node and then moves on to the next node in the sequence to retry the write.

It makes sense to me to write the metrics to the same node we just created the bucket on. Since that node just succeeded in creating a bucket, it's likely healthy (not overloaded) and able to accept the metrics successfully. Why switch to a different node to retry writing?

The InfluxDB.Write func only returns an error if there's an error from all the nodes, so it's not likely that the metrics will have to wait until the next flush interval. I'd prefer not to wait until the next flush interval anyway. Who knows if the metric buffer will run out of space and cause old metrics to be lost. It's safer to handle the error close to where it happened and also not difficult.

It also makes the code more straightforward not to have to deal with multiple cluster nodes while also dealing with missing buckets. There's no reason to have to think about the two concepts at the same time.